### PR TITLE
Change the data type of ObjectId to string from Guid.

### PR DESCRIPTION
**Pull request by adarce**
## Description
This is required for ADFS scenarios, where (unlike in AAD) object IDs are no
longer guaranteed to be Guids.

Note that this includes the AccessPolicyEntry change from the Azure SDK for .NET
NuGet package (Microsoft.Azure.Management.KeyVault.1.1.0-stack).

---

This checklist is used to make sure that common guidelines for a pull request are followed. You can find a more complete discussion of PowerShell cmdlet best practices [here](https://msdn.microsoft.com/en-us/library/dd878270(v=vs.85).aspx).

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md).**
- [x] **If changes were made to any cmdlet, the XML help was regenerated using the [platyPSHelp module](https://github.com/Azure/azure-powershell/blob/dev/documentation/platyPSHelp-documentation.md).**
- [x] **If any large changes are made to a service, they are reflected in the respective [change log](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#updating-the-change-log).**

### [General Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] The pull request does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/dev/documentation/breaking-changes.md) (unless a major version change occurs in the assembly and module).

### [Testing Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.
- [x] PowerShell scripts used in tests should do any necessary setup as part of the test or suite setup, and should not use hard-coded values for locations or existing resources.

### [Cmdlet Signature Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#cmdlet-signature-guidelines)
- [x] New cmdlets that make changes or have side effects should implement `ShouldProcess` and have `SupportShouldProcess=true` specified in the cmdlet attribute. You can find more information on `ShouldProcess` [here](https://gist.github.com/markcowl/338e16fe5c8bbf195aff9f8af0db585d#what-is-the-change).
- [x] Cmdlet specifies `OutputType` attribute if any output is produced - if the cmdlet produces no output, it should implement a `PassThru` parameter.

### [Cmdlet Parameter Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#cmdlet-parameter-guidelines)
- [x] Parameter types should not expose types from the management library - complex parameter types should be defined in the module.
- [x] Complex parameter types are discouraged - a parameter type should be simple types as often as possible. If complex types are used, they should be shallow and easily creatable from a constructor or another cmdlet.
- [x] Cmdlet parameter sets should be mutually exclusive - each parameter set must have at least one mandatory parameter not in other parameter sets.

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@adarce__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (adarce). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

**Comment by shahabhijeet** [comment]
@adarce localfeed is not an option, so update the PR once the dll is published to nuget.

**Comment by shahabhijeet** [comment]
@adarce can we get a sign off from KeyVault team.

File: src/ResourceManager/KeyVault/Commands.KeyVault/Properties/AssemblyInfo.cs [hunk 1]
```diff
@@ -29,8 +29,8 @@
 [assembly: CLSCompliant(false)]
 
 [assembly: Guid("2994548F-69B9-4DC2-8D19-52CC0C0C85BC")]
-[assembly: AssemblyVersion("1.1.3.1")]
-[assembly: AssemblyFileVersion("1.1.3.1")]
+[assembly: AssemblyVersion("2.3.0")]
+[assembly: AssemblyFileVersion("2.3.0")]
```
Comment (jkapil) [inline src/ResourceManager/KeyVault/Commands.KeyVault/Properties/AssemblyInfo.cs]: why do we need this change. We are looking for minimal changes to bring ObjectId as string

Comment (adarce) [inline src/ResourceManager/KeyVault/Commands.KeyVault/Properties/AssemblyInfo.cs]: I reverted this file.

**Comment by adarce** [comment]
@pomortaz Good catch. I've updated the target framework for both packages.config files accordingly.

**Comment by shahabhijeet** [comment]
@bganpa this PR is missing module version update as well as mark down file changes

File: src/ResourceManager/KeyVault/Commands.KeyVault/Commands/Constants.cs [hunk 2]
```diff
@@ -4,5 +4,25 @@ namespace Microsoft.Azure.Commands.KeyVault
...
+
+    public static class CmdletNoun
+    {
+        public const string AzureKeyVaultCertificate = "AzureKeyVaultCertificate";
+        public const string AzureKeyVaultCertificatePolicy = "AzureKeyVaultCertificatePolicy";
+        public const string AzureKeyVaultCertificateContact = "AzureKeyVaultCertificateContact";
+        public const string AzureKeyVaultCertificateIssuer = "AzureKeyVaultCertificateIssuer";
+        public const string AzureKeyVaultCertificateOrganizationDetails = "AzureKeyVaultCertificateOrganizationDetails";
+        public const string AzureKeyVaultCertificateAdministratorDetails = "AzureKeyVaultCertificateAdministratorDetails";
+        public const string AzureKeyVaultCertificateOperation = "AzureKeyVaultCertificateOperation";
```
Comment (adarce) [inline src/ResourceManager/KeyVault/Commands.KeyVault/Commands/Constants.cs]: This change came from the merge from the main Azure\dev branch. Note that this doesn't expose any certificate features; these are simply constant strings.

**Comment by shahabhijeet** [comment]
@bganapa @adarce I have started a on-demand run against this PR
once the on-demand run succeeds I can merged the PR.
http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/1370/

**Comment by bganapa** [comment]
@shahabhijeet on demand is successful

**Pull request merged**